pkgpanda/util.py

```diff
@@ -160,20 +160,20 @@ def _is_incomplete_download_error(exception):
     wait_random_min=1000,
     wait_random_max=2000,
     retry_on_exception=_is_incomplete_download_error)
-def _download_remote_file(out_filename, url, retries=4):
+def _download_remote_file(out_filename, url):
     with open(out_filename, "wb") as f:
         r = get_requests_retry_session().get(url, stream=True)
         r.raise_for_status()
 
-        content_length = int(r.headers['content-length'])
-
         total_bytes_read = 0
         for chunk in r.iter_content(chunk_size=4096):
             f.write(chunk)
             total_bytes_read += len(chunk)
 
-        if total_bytes_read != content_length:
-            raise IncompleteDownloadError(url, total_bytes_read, content_length)
+        if 'content-length' in r.headers:
```

Comment:
What if there is not content-length header?

Reply:
In that case we can't check whether the file size is the correct one. In order to make sure the file is correct in that case, we have to use other strategies, like checksums, using etags, etc. But that's a larger thing and we should maybe create a separate ticket for it.

But for now, in case `Content-Length` header is missing, we default back to the old behaviour, nothing's changed. See PR message here: https://github.com/dcos/dcos/pull/4629#issue-256968263